contrib/firebaseio/src/main/java/contrib/JacksonCoder.java

```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright (c) 2015 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not  use this file except  in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package contrib;
+
+import static com.google.cloud.dataflow.sdk.util.Structs.addString;
+
+import com.google.cloud.dataflow.sdk.coders.Coder;
+import com.google.cloud.dataflow.sdk.coders.CoderException;
+import com.google.cloud.dataflow.sdk.coders.StandardCoder;
+import com.google.cloud.dataflow.sdk.util.CloudObject;
+import com.google.cloud.dataflow.sdk.values.TypeDescriptor;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectMapper.DefaultTyping;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * TODO: Insert description here. (generated by elibixby)
+ */
+public class JacksonCoder<T> extends StandardCoder<T> {
+
+  private static final long serialVersionUID = -754345287170755870L;
+
+  protected static final ObjectMapper MAPPER = new ObjectMapper()
+      .enableDefaultTyping(DefaultTyping.NON_FINAL)
+      .enable(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS)
+      .enable(DeserializationFeature.WRAP_EXCEPTIONS)
+      .enable(SerializationFeature.WRAP_EXCEPTIONS)
+      .disable(SerializationFeature.FAIL_ON_UNWRAPPED_TYPE_IDENTIFIERS)
+      .enable(SerializationFeature.WRITE_NULL_MAP_VALUES);
+
+  private final ObjectWriter writer;
+  private final ObjectReader reader;
+
+  protected final Class<T> type;
+
+  public static <K> JacksonCoder<K> of(TypeDescriptor<K> type) {
+    @SuppressWarnings("unchecked")
+    Class<K> clazz = (Class<K>) type.getRawType();
+    return of(clazz);
+  }
+
+  public static <K> JacksonCoder<K> of(Class<K> clazz){
+    return new JacksonCoder<K>(clazz);
+  }
+
+  @SuppressWarnings({"unchecked", "rawtypes"})
+  @JsonCreator
+  public static JacksonCoder<?> of(
+      @JsonProperty("type") String classType) throws ClassNotFoundException {
+    return new JacksonCoder(Class.forName(classType));
+  }
+
+  public JacksonCoder(Class<T> clazz){
```

Comment:
I may be wrong, but doesn't the Jackson Serializer/Deserializer require the function annotated with `@JsonCreator` to be `public`? I'll check this.